sys/mutex.c

```diff
@@ -1,6 +1,7 @@
 #include <sync.h>
 #include <atomic.h>
 #include <mutex.h>
+#include <thread.h>
```

Comment:
I've run into some trouble with how `mutex.c` and `turnstile.h` were using features included by a long chain of indirect `#includes`. Maybe we should use IWYU or a similar tool to analyze our sources.